documentation/modules/tracing/proc-instrumenting-producers-consumers.adoc

```diff
@@ -0,0 +1,294 @@
+//
+// assembly-instrumenting-kafka-clients-tracers.adoc
+
+[id='proc-instrumenting-producers-consumers-{context}']
+= Instrumenting producers and consumers for tracing
+
+[role="_abstract"]
+Use a Decorator pattern or Interceptors to instrument your Java producer and consumer application code for tracing.
+When instrumentation is added, you can record traces when messages are produced or retrieved from a topic.
+
+The OpenTracing and OpenTelemetry instrumentation projects provide classes that support instrumentation of producers and consumers.
+You enable instrumentation in producer and consumer applications by adding the tracing JARs as a dependency to your project.
+
+Decorator instrumentation:: For Decorator instrumentation, create a modified producer or consumer instance for tracing.
+Interceptor instrumentation:: For Interceptor instrumentation, add the tracing capability to the consumer or producer configuration.
+You do this by adding the `TracingProducerInterceptor` and `TracingConsumerInterceptor` instrumentation classes.
+
+.Procedure
+
+. Add the Maven dependencies for distributed tracing to the producer or consumer's `pom.xml` file.
++
+--
+.Dependencies for OpenTracing
+[source,xml,subs="attributes+"]
+----
+<dependency>
+  <groupId>io.opentracing.contrib</groupId>
+  <artifactId>opentracing-kafka-client</artifactId>
+  <version>{OpenTracingKafkaClient}</version>
+</dependency>
+----
+.Dependencies for OpenTelemetry
+[source,xml,subs="attributes+"]
+----
+<dependency>
+	<groupId>io.opentelemetry</groupId>
+	<artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
+	<version>{OpenTelemetryVersion}</version>
+</dependency>
+<dependency>
+	<groupId>io.opentelemetry</groupId>
+	<artifactId>opentelemetry-semconv</artifactId>
+	<version>{OpenTelemetryVersion}</version>
+</dependency>
+<dependency>
+	 <groupId>io.opentelemetry</groupId>
+	<artifactId>opentelemetry-sdk-trace</artifactId>
+	<version>{OpenTelemetryVersionStable}</version>
+</dependency>
+<dependency>
+	<groupId>io.opentelemetry.instrumentation</groupId>
+	<artifactId>opentelemetry-kafka-clients-{OpenTelemetryKafkaClient}</artifactId>
+	<version>{OpenTelemetryVersion}</version>
+</dependency>
+<dependency>
+	<groupId>io.opentelemetry.instrumentation</groupId>
+	<artifactId>opentelemetry-kafka-clients-common</artifactId>
+	<version>{OpenTelemetryVersion}</version>
+</dependency>
+<dependency>
+	<groupId>io.opentelemetry</groupId>
+	<artifactId>opentelemetry-exporter-jaeger</artifactId>
+	<version>{OpenTelemetryVersionStable}</version>
+</dependency>
+<dependency>
+	<groupId>io.grpc</groupId>
+	<artifactId>grpc-netty-shaded</artifactId>
+	<version>{OpenTelemetryGrpcVersion}</version>
+</dependency>
+----
+--
+
+. Instrument your client application code using either a Decorator pattern or Interceptors.
+
+* The Decorator pattern instrumentation is different for OpenTracing and OpenTelemetry,
+but in both cases you add tracing to an instance of a producer or consumer.
```

Comment:
@alesj - Maybe we want to say more on the differences

Reply:
Current OpenTelemetry integration doesn't use existing OpenTelemetry + Kafka integartion anymore -- that interceptor.
We only get Tracer from OpenTelemetry, and then we handle tracing on our own.
Hence dunno if this is relevant anymore. @ppatierno wdyt?

Reply:
Ah, this is Operator ... hence global docs for a bit more info on how users can use OpenTelemetry with Kafka.
Then yeah, definitely leave this section.